webapp/crashstats/topcrashers/tests/test_views.py

```diff
@@ -12,259 +12,284 @@
-
+from socorro.lib.libdatetime import utc_now
+from socorro.lib.libooid import create_new_ooid
+
+
+class TestTopCrasherViews:
+    def test_topcrashers_bug_data(self, client, db, es_helper):
+        signature = "FakeSignature1"
+
+        BugAssociation.objects.create(bug_id=22222, signature=signature)
+        BugAssociation.objects.create(bug_id=33333, signature=signature)
+
+        # Index crash data for signature with associated bugs
+        crash_data = []
+        now = utc_now() - datetime.timedelta(days=1)
+        for i in range(15):
+            crash_data.append(
+                {
+                    "date_processed": now,
+                    "uuid": create_new_ooid(timestamp=now),
+                    "signature": signature,
+                    "product": "Firefox",
+                    "version": "1.0",
+                    "dom_fission_enabled": True,
+                    "is_garbage_collecting": False,
+                    "os_name": "Linux",
+                    "process_type": "parent",
+                    "startup_crash": True,
+                    "uptime": 1000,
+                }
+            )
+        for item in crash_data:
+            es_helper.index_crash(raw_crash={}, processed_crash=item, refresh=False)
+        es_helper.refresh()
 
-class TestTopCrasherViews(BaseTestViews):
-    base_url = reverse("topcrashers:topcrashers")
+        # Test Bugzilla data is rendered
+        url = reverse("topcrashers:topcrashers")
+        response = client.get(url, {"product": "Firefox", "version": "1.0"})
+        assert response.status_code == 200
+        print(response.content.decode("utf-8"))
+        doc = pyquery.PyQuery(response.content)
+        # Verify Bugzilla IDs data
+        bug_ids = [x.text for x in doc("td.bug_ids_more > a")]
+        # Verify bugs are sorted by bug id descending
+        assert bug_ids == ["33333", "22222"]
 
-    def test_topcrashers(self):
-        BugAssociation.objects.create(bug_id=123456789, signature="Something")
-        BugAssociation.objects.create(
-            bug_id=22222, signature="FakeSignature1 \u7684 Japanese"
-        )
-        BugAssociation.objects.create(
-            bug_id=33333, signature="FakeSignature1 \u7684 Japanese"
-        )
+    def test_topcrashers_first_appearance(self, client, db, es_helper):
+        signature = "FakeSignature1"
 
         Signature.objects.create(
-            signature="FakeSignature1 \u7684 Japanese",
-            first_date=datetime.datetime(2000, 1, 1, 12, 23, 34, tzinfo=utc),
+            signature=signature,
+            first_date=datetime.datetime(2021, 1, 1, 12, 23, 34, tzinfo=utc),
             first_build="20000101122334",
         )
-        Signature.objects.create(
-            signature="mozCool()",
-            first_date=datetime.datetime(2016, 5, 2, 0, 0, 0, tzinfo=utc),
-            first_build="20160502000000",
-        )
 
-        def mocked_supersearch_get(**params):
```

Comment:
This mocked output was wrong in a bunch of weird ways, so it's not clear what these tests were telling us about correctness in the system.